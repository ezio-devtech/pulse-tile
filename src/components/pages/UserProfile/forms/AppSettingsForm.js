import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
