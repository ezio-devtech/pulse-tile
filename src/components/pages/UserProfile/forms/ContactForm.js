import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
