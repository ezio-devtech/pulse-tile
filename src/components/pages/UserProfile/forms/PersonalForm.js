import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from './validation';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="First Name"
                      name="firstname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Last Name"
                      name="lastname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="NHS No"
                      name="nhs"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Date of Birth"
                      name="birthday"
                      type="text"
                      placeholder=""
                      component={DateInput}
                    />
                    <Field
                      label="Gender"
                      name="gender"
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label="Doctor"
                      name="doctor"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
