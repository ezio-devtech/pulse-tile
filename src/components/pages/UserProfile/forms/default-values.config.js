import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from './values-names.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
