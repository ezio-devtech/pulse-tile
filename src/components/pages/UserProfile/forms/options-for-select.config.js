const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
