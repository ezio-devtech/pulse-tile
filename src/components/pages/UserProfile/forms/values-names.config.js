export const valuesNames = {
  APP_TITLE: 'title',
  LOGO_PATH: 'logoPath',
  SELECT_THEME: 'selectThemeField',
  BROWSER_TITLE: 'browserWindowTitle',
};
