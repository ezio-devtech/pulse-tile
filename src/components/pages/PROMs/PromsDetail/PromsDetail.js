import React, { PureComponent } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import _ from 'lodash/fp';

import PluginDetailPanel from '../../../plugin-page-component/PluginDetailPanel';
import PromsDetailForm from './PromsDetailForm';
import FormTitle from '../../../ui-elements/FormTitle/FormTitle';
import RecordsOfTableView from '../../../form-fields/RecordsOfTable/RecordsOfTableView';
import { getDDMMMYYYY } from '../../../../utils/time-helpers.utils';
import { valuesNames, valuesLabels, qestionOptions } from '../forms.config';
import { patientProceduresDetailSelector } from '../../Procedures/selectors';
import { fetchPatientProceduresDetailRequest } from '../../Procedures/ducks/fetch-patient-procedures-detail.duck';

const PROM_PANEL = 'promPanel';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ fetchPatientProceduresDetailRequest }, dispatch) });

@connect(patientProceduresDetailSelector, mapDispatchToProps)
export default class PromsDetail extends PureComponent {
  componentDidMount() {
    this.getProcedureById(this.props.detail, this.props.procedureDetail);
  }

  componentWillReceiveProps(nextProps) {
    this.getProcedureById(nextProps.detail, nextProps.procedureDetail);
  }

  getProcedureById = (detail, procedure) => {
    const { actions, match } = this.props;
    const userId = _.get('params.userId', match);

    if (userId && detail && detail[valuesNames.PROCEDURE_ID]) {
      let procedureId = detail[valuesNames.PROCEDURE_ID];
      procedureId = procedureId.replace(/ehr:\/\/\//, '');
      procedureId = procedureId.replace(/::.*$/, '');

      if (!procedure || procedure.sourceId !== procedureId) {
        actions.fetchPatientProceduresDetailRequest({ userId, sourceId: procedureId });
      }
    }
  };

  getProcedureRecord = procedure => ([{
    date: getDDMMMYYYY(procedure.date),
    name: procedure.name,
    source: procedure.source,
    sourceId: procedure.sourceId,
    type: 'procedures',
    typeTitle: 'Procedures',
  }]);

  render() {
    const { onExpand, onShow, openedPanel, expandedPanel, currentPanel, onEdit, editedPanel, onCancel, onSaveSettings, promsDetailFormValues, isSubmit, match, status, changeScoreStatus } = this.props;
    let { detail, procedureDetail } = this.props;
    detail = detail || {};

    if (detail[valuesNames.PROCEDURE_ID] && procedureDetail) {
      detail[valuesNames.RECORDS] = this.getProcedureRecord(procedureDetail);
    }

    return (
      <div className="section-detail">
        <div className="panel-group accordion">
          {(expandedPanel === PROM_PANEL || expandedPanel === 'all') && !editedPanel[PROM_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={PROM_PANEL}
            title="PROM"
            onShow={onShow}
            isOpen={openedPanel === PROM_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={promsDetailFormValues}
          >
            <div className="panel-body-inner">
              <div className="form">
                <div className="form-group-wrapper">
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.NAME}</label>
                        <div className="form-control-static">{detail[valuesNames.NAME]}</div>
                      </div>
                    </div>
                  </div>

                  <RecordsOfTableView records={detail[valuesNames.RECORDS]} />

                  <FormTitle text="Specific Question" />
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.QUESTION_1}</label>
                        <div className="form-control-static">{qestionOptions[detail[valuesNames.SPECIFIC_Q1]]}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.QUESTION_2}</label>
                        <div className="form-control-static">{qestionOptions[detail[valuesNames.SPECIFIC_Q2]]}</div>
                      </div>
                    </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.QUESTION_3}</label>
                        <div className="form-control-static">{qestionOptions[detail[valuesNames.SPECIFIC_Q3]]}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.QUESTION_4}</label>
                        <div className="form-control-static">{qestionOptions[detail[valuesNames.SPECIFIC_Q4]]}</div>
                      </div>
                    </div>
                  </div>

                  <div className="col-expand-right">
                    <div className="form-group">
                      <label className="control-label">{valuesLabels.QUESTION_5}</label>
                      <div className="form-control-static">{qestionOptions[detail[valuesNames.SPECIFIC_Q5]]}</div>
                    </div>
                  </div>

                  <FormTitle text="General Score" />
                  <div className="form-group">
                    <div>Pain severity on a scale of 0 to 100, where 0 indicates no pain and 100 indicates severe pain.</div>
                  </div>
                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="highlighter-input-group highlighter-wrapper">
                        <span className={`highlighter-${status}`} />
                        <label className="control-label">{valuesLabels.SCORE}</label>
                        <div className={`input-holder highlighter-input-holder ${status}`}>
                          <div className="form-control input-sm">{detail[valuesNames.SCORE]}</div>
                        </div>
                      </div>
                    </div>
                  </div>

                  <div className="row-expand">
                    <div className="col-expand-left">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.DATE_CREATED}</label>
                        <div className="form-control-static">{getDDMMMYYYY(detail[valuesNames.DATE_CREATED])}</div>
                      </div>
                    </div>
                    <div className="col-expand-right">
                      <div className="form-group">
                        <label className="control-label">{valuesLabels.SOURCE}</label>
                        <div className="form-control-static">{detail[valuesNames.SOURCE]}</div>
                      </div>
                    </div>
                  </div>

                </div>
              </div>
            </div>
          </PluginDetailPanel> : null}

          {(expandedPanel === PROM_PANEL || expandedPanel === 'all') && editedPanel[PROM_PANEL] ? <PluginDetailPanel
            onExpand={onExpand}
            name={PROM_PANEL}
            title="Edit PROM"
            onShow={onShow}
            isOpen={openedPanel === PROM_PANEL}
            currentPanel={currentPanel}
            onEdit={onEdit}
            editedPanel={editedPanel}
            onCancel={onCancel}
            onSaveSettings={onSaveSettings}
            formValues={promsDetailFormValues}
            isBtnShowPanel
          >
            <PromsDetailForm
              detail={detail}
              isSubmit={isSubmit}
              match={match}
              status={status}
              changeScoreStatus={changeScoreStatus}
            />
          </PluginDetailPanel> : null }
        </div>
      </div>
    )
  }
}
