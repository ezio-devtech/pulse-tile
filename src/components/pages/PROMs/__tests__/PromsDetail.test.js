import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';

import PromsDetail from '../PromsDetail/PromsDetail';
import { valuesNames, valuesLabels, qestionOptions } from '../forms.config';

Enzyme.configure({ adapter: new Adapter() });

Date.now = jest.fn(() => new Date(Date.UTC(2018, 1, 1, 1)).valueOf());

// storage setup in different configurations
const mockStore = configureStore();

const propsForPromsPanel = {
  detail: {
    [valuesNames.NAME]: 'test Proms 1',
    [valuesNames.RECORDS]: [
      {
        [valuesNames.RECORDS_DATE]: 1482190593395,
        [valuesNames.RECORDS_NAME]: 'test records',
        [valuesNames.SOURCE]: 'test records source',
        [valuesNames.SOURCE_ID]: 'test records sourceId',
        type: 'test records type',
        [valuesNames.TYPE]: 'test records typeTitle',
      },
    ],
    [valuesNames.SCORE]: '90',
    [valuesNames.DATE_CREATED]: 1482170593395,
    [valuesNames.SPECIFIC_Q1]: 'at0009',
    [valuesNames.SPECIFIC_Q2]: 'at0015',
    [valuesNames.SPECIFIC_Q3]: 'at0021',
    [valuesNames.SPECIFIC_Q4]: 'at0025',
    [valuesNames.SPECIFIC_Q5]: 'at0030',
    [valuesNames.AUTHOR]: 'DR Mary Jones',
    [valuesNames.SOURCE]: 'openehr',
    [valuesNames.SOURCE_ID]: 'testSourceID1',
    [valuesNames.PROCEDURE_ID]: 'testSourceID1',
  },
};

const store = mockStore({
  proceduresDetail: {
    '9999999000': {
      [valuesNames.NAME]: '123b',
      [valuesNames.PROCEDURE_NAME]: '123b',
      [valuesNames.CODE]: 'at0039',
      [valuesNames.DATE_OF_PROCEDURE]: 1510579904000,
      [valuesNames.TIME]: 48704000,
      [valuesNames.TERMINOLOGY]: 'SNOMED-CT',
      [valuesNames.NOTES]: '123b',
      [valuesNames.PERFORMER]: 'Performer',
      [valuesNames.STATUS]: 'completed',
      [valuesNames.AUTHOR]: 'bob.smith@gmail.com',
      [valuesNames.DATE]: 1510659104000,
      [valuesNames.SOURCE]: 'ethercis',
      [valuesNames.SOURCE_ID]: '08102813-e8d0-4f45-aa95-ac0d633e69fd',
    }
  }
});

const match = {
  params: {
    userId: '9999999000',
  },
};

const PROM_PANEL = 'promPanel';

describe('Component <PromsDetail />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(<PromsDetail
      store={store}
      match={match}
    />).dive();

    // Testing component when detail filled object, expandedPanel is all, and panel not edited
    component.setProps({ detail: propsForPromsPanel.detail, expandedPanel: 'all', editedPanel: { [PROM_PANEL]: false } });
    expect(component.props().className).toEqual('section-detail');
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('RecordsOfTableView')).toHaveLength(1);

    // Testing promPanel
    expect(component.find('PluginDetailPanel').at(0).props().name).toEqual(PROM_PANEL);
    expect(component.find('PluginDetailPanel').at(0).props().title).toEqual('PROM');
    expect(component.find('PluginDetailPanel').at(0).props().isOpen).toEqual(false);
    expect(component.find('PluginDetailPanel').at(0).props().isShowControlPanel).toEqual(true);

    expect(component.find('RecordsOfTableView').props().records).toEqual(propsForPromsPanel.detail[valuesNames.RECORDS]);

    expect(component.find('.control-label').at(0).text()).toEqual(valuesLabels.NAME);
    expect(component.find('.control-label').at(1).text()).toEqual(valuesLabels.QUESTION_1);
    expect(component.find('.control-label').at(2).text()).toEqual(valuesLabels.QUESTION_2);
    expect(component.find('.control-label').at(3).text()).toEqual(valuesLabels.QUESTION_3);
    expect(component.find('.control-label').at(4).text()).toEqual(valuesLabels.QUESTION_4);
    expect(component.find('.control-label').at(5).text()).toEqual(valuesLabels.QUESTION_5);
    expect(component.find('.control-label').at(6).text()).toEqual(valuesLabels.SCORE);
    expect(component.find('.control-label').at(7).text()).toEqual(valuesLabels.DATE_CREATED);
    expect(component.find('.control-label').at(8).text()).toEqual(valuesLabels.SOURCE);

    expect(component.find('.form-control-static').at(0).text()).toEqual(propsForPromsPanel.detail[valuesNames.NAME]);
    expect(component.find('.form-control-static').at(1).text()).toEqual(qestionOptions[propsForPromsPanel.detail[valuesNames.SPECIFIC_Q1]]);
    expect(component.find('.form-control-static').at(2).text()).toEqual(qestionOptions[propsForPromsPanel.detail[valuesNames.SPECIFIC_Q2]]);
    expect(component.find('.form-control-static').at(3).text()).toEqual(qestionOptions[propsForPromsPanel.detail[valuesNames.SPECIFIC_Q3]]);
    expect(component.find('.form-control-static').at(4).text()).toEqual(qestionOptions[propsForPromsPanel.detail[valuesNames.SPECIFIC_Q4]]);
    expect(component.find('.form-control-static').at(5).text()).toEqual(qestionOptions[propsForPromsPanel.detail[valuesNames.SPECIFIC_Q5]]);
    expect(component.find('.form-control-static').at(7).text()).toEqual(propsForPromsPanel.detail[valuesNames.SOURCE]);

    expect(component.find('.form-control').at(0).text()).toEqual(propsForPromsPanel.detail[valuesNames.SCORE]);

    expect(component).toMatchSnapshot();

    component.setProps({ detail: {} });
    expect(component.find('RecordsOfTableView')).toHaveLength(1);
    expect(component).toMatchSnapshot();

  });

  it('should renders correctly with different state of props', () => {
    const component = shallow(<PromsDetail
        store={store}
        match={match}
      />).dive();
    // Testing component when detail empty object, expandedPanel is promPanel
    component.setProps({ detail: { [valuesNames.DATE_CREATED]: 1507020019000 }, expandedPanel: PROM_PANEL, editedPanel: { [PROM_PANEL]: false } });
    expect(component.find('PluginDetailPanel')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(PROM_PANEL);
    expect(component).toMatchSnapshot();

    // Testing component when detail filled object, expandedPanel is all, and panel is edited
    component.setProps({ detail: propsForPromsPanel.detail, expandedPanel: PROM_PANEL, editedPanel: { [PROM_PANEL]: true } });
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('PluginDetailPanel').props().name).toEqual(PROM_PANEL);
    expect(component).toMatchSnapshot();
  });
});

