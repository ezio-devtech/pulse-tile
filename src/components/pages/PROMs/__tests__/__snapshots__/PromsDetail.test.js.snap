// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <PromsDetail /> should renders correctly with different state of props 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "promPanel": false,
        }
      }
      isOpen={false}
      isShowControlPanel={true}
      name="promPanel"
      title="PROM"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Name of PROM
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <RecordsOfTableView />
            <FormTitle
              text="Specific Question"
            />
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    1. In general, would you say your health is:
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    2. Compared to one year ago, how would you rate your health in general now?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    3. Does your health now limit you in vigorous activities?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    4. Does your health now limit you in moderate activities?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  5. Does your health now limit you in lifting or carrying groceries?
                </label>
                <div
                  className="form-control-static"
                />
              </div>
            </div>
            <FormTitle
              text="General Score"
            />
            <div
              className="form-group"
            >
              <div>
                Pain severity on a scale of 0 to 100, where 0 indicates no pain and 100 indicates severe pain.
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="highlighter-input-group highlighter-wrapper"
                >
                  <span
                    className="highlighter-undefined"
                  />
                  <label
                    className="control-label"
                  >
                    Score
                  </label>
                  <div
                    className="input-holder highlighter-input-holder undefined"
                  >
                    <div
                      className="form-control input-sm"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    03-Oct-2017
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <PromsDetail /> should renders correctly with different state of props 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "promPanel": true,
        }
      }
      isBtnShowPanel={true}
      isOpen={false}
      isShowControlPanel={true}
      name="promPanel"
      title="Edit PROM"
    >
      <ReduxForm
        detail={
          Object {
            "author": "DR Mary Jones",
            "dateCreated": 1482170593395,
            "name": "test Proms 1",
            "procedureSourceId": "testSourceID1",
            "records": Array [
              Object {
                "date": "01-Feb-2018",
                "name": "123b",
                "source": "ethercis",
                "sourceId": "08102813-e8d0-4f45-aa95-ac0d633e69fd",
                "type": "procedures",
                "typeTitle": "Procedures",
              },
            ],
            "score": "90",
            "source": "openehr",
            "sourceId": "testSourceID1",
            "specific_q1": "at0009",
            "specific_q2": "at0015",
            "specific_q3": "at0021",
            "specific_q4": "at0025",
            "specific_q5": "at0030",
          }
        }
        match={
          Object {
            "params": Object {
              "userId": "9999999000",
            },
          }
        }
      />
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <PromsDetail /> should renders with props correctly 1`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "promPanel": false,
        }
      }
      isOpen={false}
      isShowControlPanel={true}
      name="promPanel"
      title="PROM"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Name of PROM
                  </label>
                  <div
                    className="form-control-static"
                  >
                    test Proms 1
                  </div>
                </div>
              </div>
            </div>
            <RecordsOfTableView
              records={
                Array [
                  Object {
                    "date": "01-Feb-2018",
                    "name": "123b",
                    "source": "ethercis",
                    "sourceId": "08102813-e8d0-4f45-aa95-ac0d633e69fd",
                    "type": "procedures",
                    "typeTitle": "Procedures",
                  },
                ]
              }
            />
            <FormTitle
              text="Specific Question"
            />
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    1. In general, would you say your health is:
                  </label>
                  <div
                    className="form-control-static"
                  >
                    Excellent
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    2. Compared to one year ago, how would you rate your health in general now?
                  </label>
                  <div
                    className="form-control-static"
                  >
                    Much better
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    3. Does your health now limit you in vigorous activities?
                  </label>
                  <div
                    className="form-control-static"
                  >
                    Yes, limited a lot
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    4. Does your health now limit you in moderate activities?
                  </label>
                  <div
                    className="form-control-static"
                  >
                    Yes, limited a lot
                  </div>
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  5. Does your health now limit you in lifting or carrying groceries?
                </label>
                <div
                  className="form-control-static"
                >
                  Yes, limited a lot
                </div>
              </div>
            </div>
            <FormTitle
              text="General Score"
            />
            <div
              className="form-group"
            >
              <div>
                Pain severity on a scale of 0 to 100, where 0 indicates no pain and 100 indicates severe pain.
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="highlighter-input-group highlighter-wrapper"
                >
                  <span
                    className="highlighter-undefined"
                  />
                  <label
                    className="control-label"
                  >
                    Score
                  </label>
                  <div
                    className="input-holder highlighter-input-holder undefined"
                  >
                    <div
                      className="form-control input-sm"
                    >
                      90
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    19-Dec-2016
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  >
                    openehr
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;

exports[`Component <PromsDetail /> should renders with props correctly 2`] = `
<div
  className="section-detail"
>
  <div
    className="panel-group accordion"
  >
    <PluginDetailPanel
      editedPanel={
        Object {
          "promPanel": false,
        }
      }
      isOpen={false}
      isShowControlPanel={true}
      name="promPanel"
      title="PROM"
    >
      <div
        className="panel-body-inner"
      >
        <div
          className="form"
        >
          <div
            className="form-group-wrapper"
          >
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Name of PROM
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <RecordsOfTableView />
            <FormTitle
              text="Specific Question"
            />
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    1. In general, would you say your health is:
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    2. Compared to one year ago, how would you rate your health in general now?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    3. Does your health now limit you in vigorous activities?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    4. Does your health now limit you in moderate activities?
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
            <div
              className="col-expand-right"
            >
              <div
                className="form-group"
              >
                <label
                  className="control-label"
                >
                  5. Does your health now limit you in lifting or carrying groceries?
                </label>
                <div
                  className="form-control-static"
                />
              </div>
            </div>
            <FormTitle
              text="General Score"
            />
            <div
              className="form-group"
            >
              <div>
                Pain severity on a scale of 0 to 100, where 0 indicates no pain and 100 indicates severe pain.
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="highlighter-input-group highlighter-wrapper"
                >
                  <span
                    className="highlighter-undefined"
                  />
                  <label
                    className="control-label"
                  >
                    Score
                  </label>
                  <div
                    className="input-holder highlighter-input-holder undefined"
                  >
                    <div
                      className="form-control input-sm"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row-expand"
            >
              <div
                className="col-expand-left"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Date
                  </label>
                  <div
                    className="form-control-static"
                  >
                    01-Feb-2018
                  </div>
                </div>
              </div>
              <div
                className="col-expand-right"
              >
                <div
                  className="form-group"
                >
                  <label
                    className="control-label"
                  >
                    Source
                  </label>
                  <div
                    className="form-control-static"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </PluginDetailPanel>
  </div>
</div>
`;
